inotify.go

```diff
@@ -101,9 +125,14 @@ func (w *Watcher) Add(name string) error {
 
 // Remove stops watching the the named file or directory (non-recursively).
 func (w *Watcher) Remove(name string) error {
-	name = filepath.Clean(name)
 	w.mu.Lock()
 	defer w.mu.Unlock()
+	return w.remove(name)
+}
+
+// Only call this when w.mu is locked, like from Remove or Close.
```

Comment:
I'm not in love with having a function that requires the mutex to be done outside of it, but I don't have any better ideas right now.

Also not sure about taking a Mutex around the entirety of remove vs. just some part of it.